packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -1839,3 +1841,262 @@ export function openClose(componentTagOrHTML: TagOrHTML, options?: OpenCloseOpti
     });
   }
 }
+
+/**
+ *
+ * Helper to test custom theming of a component's associated tokens.
+ *
+ * @example
+ * describe("theme", () => {
+ *   const tokens = {
+ *     "--calcite-action-bar-trigger-background-color": [{
+ *       selector: "calcite-action-bar",
+ *       targetProp: "backgroundColor",
+ *     }, {
+ *       selector: "calcite-action-bar",
+ *       shadowSelector: "calcite-action-group calcite-action >>> .button",
+ *       targetProp: "backgroundColor",
+ *     }],
+ *     "--calcite-action-bar-trigger-background-color-active": {
+ *       selector: "calcite-action-bar",
+ *       shadowSelector: "calcite-action-group calcite-action >>> .button",
+ *       targetProp: "backgroundColor",
+ *       state: { press: { attribute: "class", value: CSS.expandToggle } },
+ *     },
+ *     "--calcite-action-bar-trigger-background-color-focus": {
+ *       selector: "calcite-action-bar",
+ *       shadowSelector: "calcite-action-group calcite-action >>> .button",
+ *       targetProp: "backgroundColor",
+ *       state: "focus",
+ *     },
+ *     "--calcite-action-bar-trigger-background-color-hover": {
+ *       selector: "calcite-action-bar",
+ *       shadowSelector: "calcite-action-group calcite-action >>> .button",
+ *       targetProp: "backgroundColor",
+ *       state: "hover",
+ *     },
+ *   };
+ *   themed(`calcite-action-bar`, tokens);
+ * });
+ *
+ * @param componentTagOrHTML  - The component tag or HTML markup to test against.
+ * @param tokens - A record of token names and their associated selectors, shadow selectors, target props, and states.
+ */
+export function themed(
+  componentTagOrHTML: TagOrHTML,
+  tokens: Record<string, TestSelectToken | TestSelectToken[]>,
+): void {
+  it("is themeable", async () => {
+    const page = await simplePageSetup(componentTagOrHTML);
+    const setTokens: Record<string, string> = {};
+    const styleTargets: Record<string, [E2EElement, string[]]> = {};
+    const testTargets: TestTarget[] = [];
+
+    // Parse test config for tokens and selectors
+    for (const token in tokens) {
+      let selectors = tokens[token];
+
+      if (!isArray(selectors)) {
```

Comment:
Can you use `Array.isArray` instead of adding a custom function for this?

Reply:
no. The custom type guard is required. See https://github.com/microsoft/TypeScript/issues/19892

Reply:
Are you seeing incorrect types or a particular TS error? The issue you referenced is related to `ReadOnlyArray` type, which we are not using.

Types seem fine in both code paths.

![Screenshot 2024-04-05 at 11 05 56 AM](https://github.com/Esri/calcite-design-system/assets/197440/5ada8e42-14a8-41c8-9750-26ad8e8b100d)

![Screenshot 2024-04-05 at 11 11 24 AM](https://github.com/Esri/calcite-design-system/assets/197440/5757b369-fbdd-4fa0-b4a8-c70e47ad6c16)

Reply:
That doesn't work for me. It says it doesn't know if it's a string or an array.